src/oscar/apps/dashboard/orders/views.py

```diff
@@ -356,43 +367,40 @@ def render_to_response(self, context, **response_kwargs):
     def get_download_filename(self, request):
         return 'orders.csv'
 
+    @cached_property
+    def sorted_csv_columns(self):
+        columns = OrderedDict()
+        for k, v in self.CSV_COLUMNS:
+            columns[k] = v
+        return columns
+
+    def get_row_values(self, order):
+        row = self.sorted_csv_columns.copy()
```

Comment:
I know the old code was doing this, but I think it's confusing to initialise the `row` with incorrect (heading) values. I think this method should return a regular dict - we should then order them before rendering the row (by looping over the columns).

Reply:
(This should also mean we only have to read `self.CSV_COLUMNS` once, in the code that generates the CSV.